### ISSUE TITLE
Compilation currently broken

### ISSUE THREAD

**Issue by alexandrebouchard** [UBC-Stat-ML/blangSDK#173]
~~See https://github.com/google/guava/issues/7132~~

**Comment by alexandrebouchard** [comment]
https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2252

**Comment by alexandrebouchard** [comment]
Example:

```
./gradlew build
```
gives:
```
java.lang.UnsupportedClassVersionError: org/eclipse/core/runtime/OperationCanceledException 
has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 55.0
```

**Comment by alexandrebouchard** [comment]
While trying to upgrade to 17 seems a large undertaking, tried many times before. E.g. just starting with gradle scripts, they do not support 17, and upgrading gradle creates many errors.

**Comment by alexandrebouchard** [comment]
Got more info with `-d` for debug mode

```
./gradlew -d build | grep core | grep eclipse
```

just before the crash:
```
2024-03-29T21:27:07.233-0700 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.RepositoryChainComponentMetaDataResolver] Using org.eclipse.emf:org.eclipse.emf.ecore:2.20.0 from Maven repository 'MavenRepo'
```
So current suspect is culpit is `org.eclipse.emf:org.eclipse.emf.ecore:2.20.0`

**Comment by alexandrebouchard** [comment]
Idea: force downgrading it: 

https://docs.gradle.org/6.8.3/userguide/dependency_downgrade_and_exclude.html

Example:

```
dependencies {
    implementation 'org.apache.httpcomponents:httpclient:4.5.4'
    implementation('commons-codec:commons-codec') {
        version {
            strictly '1.9'
        }
    }
}
```

**Comment by alexandrebouchard** [comment]
Datapoint: also fails with `./setup-eclipse.sh` 

Perhaps problem is with the plugin?

**Comment by alexandrebouchard** [comment]
Digging into the gradle cache to find culpits...

version-of-jar.sh:
```
#!/bin/bash

rm -r ~/tmp/decompress
mkdir ~/tmp/decompress
cp $1 ~/tmp/decompress 
cd ~/tmp/decompress 
unzip -qq *.jar
grep Bundle-RequiredExecutionEnvironment META-INF/MANIFEST.MF 
cd - > /dev/null
```

version-of-jars.sh:
```
#!/bin/bash

for i in `find . -type f -name "*.jar"`
do
	echo $i --- `version-of-jar.sh $i`
done
```

Deleted .gradle, then running once, then
From `~.gradle/caches/modules-2/files-2.1`, 

```
version-of-jars.sh  | grep JavaSE-17
```

got:

```
./org.eclipse.platform/org.eclipse.equinox.registry/3.12.0/d873328963eee04833540334b9868a4b5894f1fa/org.eclipse.equinox.registry-3.12.0.jar --- Bundle-RequiredExecutionEnvironment: JavaSE-17
./org.eclipse.platform/org.eclipse.core.runtime/3.31.0/b815a6f4da0be97526bfe7d915054e80f2bc8197/org.eclipse.core.runtime-3.31.0.jar --- Bundle-RequiredExecutionEnvironment: JavaSE-17
./org.eclipse.platform/org.eclipse.equinox.app/1.7.0/27ad31338c942b3b8ff693780b99514f25257940/org.eclipse.equinox.app-1.7.0.jar --- Bundle-RequiredExecutionEnvironment: JavaSE-17
./org.eclipse.platform/org.eclipse.core.resources/3.20.100/6dd19c7161f9d5965394f70d7d32ff8238bdff6f/org.eclipse.core.resources-3.20.100.jar --- Bundle-RequiredExecutionEnvironment: JavaSE-17
./org.eclipse.platform/org.eclipse.equinox.preferences/3.11.0/94a74fb01448da1df6a0f7e896bdd2b337683745/org.eclipse.equinox.preferences-3.11.0.jar --- Bundle-RequiredExecutionEnvironment: JavaSE-17
./org.eclipse.platform/org.eclipse.equinox.common/3.19.0/5b6633ba85fc1c59a6024e1068a2a0a39458445d/org.eclipse.equinox.common-3.19.0.jar --- Bundle-RequiredExecutionEnvironment: JavaSE-17
./org.eclipse.platform/org.eclipse.core.commands/3.12.0/d1604534d74b8119e905f58b929f3b10940c43be/org.eclipse.core.commands-3.12.0.jar --- Bundle-RequiredExecutionEnvironment: JavaSE-17
./org.eclipse.platform/org.eclipse.core.jobs/3.15.200/6d32405156e9732768b3374bfb2bdcc4f49f0c54/org.eclipse.core.jobs-3.15.200.jar --- Bundle-RequiredExecutionEnvironment: JavaSE-17
./org.eclipse.platform/org.eclipse.text/3.14.0/8e3578da612c813d7bb062a0a6d548b948b20a8d/org.eclipse.text-3.14.0.jar --- Bundle-RequiredExecutionEnvironment: JavaSE-17
./org.eclipse.platform/org.eclipse.core.contenttype/3.9.300/a4670d530c7b4c2ba6d2374947d7e8bdb027d2ac/org.eclipse.core.contenttype-3.9.300.jar --- Bundle-RequiredExecutionEnvironment: JavaSE-17
./org.eclipse.platform/org.eclipse.core.expressions/3.9.300/62eeac77a1db70866cab80d3948052cd472bca82/org.eclipse.core.expressions-3.9.300.jar --- Bundle-RequiredExecutionEnvironment: JavaSE-17
./org.eclipse.platform/org.eclipse.core.filesystem/1.10.300/9bc848bd15920f3f01336496fee25c4fb2095a5f/org.eclipse.core.filesystem-1.10.300.jar --- Bundle-RequiredExecutionEnvironment: JavaSE-17
```

So we have narrowed down the exact culpits. Now who import those?

**Comment by alexandrebouchard** [comment]
From `./gradlew dependencies`, it is clear that e.g. org.eclipse.core.runtime gets resolved to 3.7.0, not 3.31.0. 

HOWEVER, on Maven Central, the earliest version available is 3.12.0, so maybe old versions got purged. 

https://repo1.maven.org/maven2/org/eclipse/platform/org.eclipse.core.runtime/

https://mvnrepository.com/artifact/org.eclipse.platform/org.eclipse.core.runtime

**Comment by alexandrebouchard** [comment]
For future occurrence of something like this, we now have an internal back up of a full gradle cache (working-gradle-cache-for-blang-sdk.zip)
